src/autocomplete/EmojiProvider.tsx

```diff
@@ -67,7 +67,7 @@ export default class EmojiProvider extends AutocompleteProvider {
     constructor() {
         super(EMOJI_REGEX);
         this.matcher = new QueryMatcher<ISortedEmoji>(SORTED_EMOJI, {
-            keys: ['emoji.emoticon'],
+            keys: [],
```

Comment:
question: does this class still do anything with an empty list of keys?

Reply:
It does [this](https://github.com/matrix-org/matrix-react-sdk/blob/5344b86de422cfe35e4a5cafc57b9667beed2467/src/autocomplete/QueryMatcher.ts#L68), so basically not much

Reply:
Could we remove `this.matcher`?

Reply:
Sorry, I don't think I know what you mean... Why would you want to remove that?

Reply:
My original question was to try to establish what the remaining purpose of `this.matcher` is - now that `keys` is empty - and if it now does not serve a purpose, to remove the object).

(bearing in mind I don't understand this code very well, and it may well still serve a purpose!)

Reply:
Well, the `QueryMatcher` is the thing that does the autocomplete searching, so we can't exist without it. We are also passing in the `funcs` param which serves almost the same function as do the `keys`